### ISSUE TITLE
Error with modified geometry & count of photons

### ISSUE THREAD

**Issue by Qudzu** [danieljfarrell/pvtrace#23]
Dear Daniel,
First and foremost, thank you for the PVTRACE 2.0 
I am now working on a small simulation, and I noticed some problems that I cannot solve.
1) I keep getting the following error, whenever I try to simulate a box geometry with modified dimensions (for example, rectancular one):

> PYDEV DEBUGGER WARNING:
sys.settrace() should not be used when the debugger is being used.
This may cause the debugger to stop working correctly.
If this is needed, please check:
http://pydev.blogspot.com/2007/06/why-cant-pydev-debugger-work-with.html
to see how to restore the debug tracing back correctly.
Call Location:
  File "C:\Users\jko\AppData\Local\Continuum\anaconda3\lib\bdb.py", line 332, in set_trace
    sys.settrace(self.trace_dispatch)
> c:\users\jko\appdata\local\continuum\anaconda3\lib\site-packages\pvtrace\material\host.py(92)trace_surface()
-> if not all([isinstance(x, Refractive) for x in (container_geometry.material, to_geometry.material)]):

To show it, I use slightly modified version of "LSC" example you provided: 

```
import numpy as np
import pathlib
import time
import sys
import os
import matplotlib
import pandas as pd
from dataclasses import asdict
import matplotlib
import matplotlib.pyplot as plt
import functools
from pvtrace.light.utils import wavelength_to_rgb
from pvtrace.scene.scene import Scene
from pvtrace.algorithm import photon_tracer
from pvtrace.scene.node import Node
from pvtrace.common.errors import TraceError
from pvtrace.light.light import Light
from pvtrace.light.ray import Ray
from pvtrace.geometry.utils import EPS_ZERO
from pvtrace.geometry.mesh import Mesh
from pvtrace.geometry.box import Box
from pvtrace.geometry.sphere import Sphere
from pvtrace.material.dielectric import Dielectric
from pvtrace.material.lumophore import Lumophore
from pvtrace.material.host import Host
from pvtrace.scene.node import Node
from pvtrace.scene.renderer import MeshcatRenderer
from pvtrace.geometry.utils import magnitude
import logging
# We want to see pvtrace logging here
logging.getLogger('pvtrace').setLevel(logging.CRITICAL)
logging.getLogger('trimesh').setLevel(logging.CRITICAL)
logging.getLogger('matplotlib').setLevel(logging.CRITICAL)

wavelength_range = (200, 800)
wavelength = np.linspace(*wavelength_range, 1000)
lumogen = Lumophore.make_lumogen_f_red(wavelength, 20, 1.0)
linear_background = Lumophore.make_linear_background(wavelength, 1.0)

# Make a world coordinate system
world_node = Node(name='world')
world_node.geometry = Sphere(
    radius=10.0,
    material=Dielectric.make_constant((300, 1000.0), 1.0)
)

refractive_index = np.column_stack(
    (wavelength, np.ones(wavelength.shape) * 1.5)
)
# Add LSC
size = (5.0, 1.0, 0.02)
lsc = Node(name="LSC", parent=world_node)
lsc.geometry = Box(
    size, 
    material=Host(
        refractive_index,  # LSC refractive index
        [linear_background, lumogen]  # LSC list of lumophore materials
    )
)

# Light source
light = Light(
    divergence_delegate=functools.partial(
        Light.cone_divergence,
        np.radians(20)
    )
)
light_node = Node(
    name='light',
    parent=world_node,
    location=(0.0, 0.0, 1.0)
)
light_node.rotate(np.radians(180), (1, 0, 0))
light_node.light = light
scene = Scene(root=world_node)

renderer = MeshcatRenderer(max_histories=None)
renderer.render(scene)
renderer.vis.jupyter_cell()

entrance_rays = []
exit_rays = []
max_rays = 200

renderer.vis.delete()
renderer.render(scene)
for light_node in scene.light_nodes:
    for idx, ray in enumerate(light.emit(max_rays)):
        ray = ray.representation(light_node, world_node)
        try:
            steps = photon_tracer.follow(ray, scene)
            path, decisions = zip(*steps)
            renderer.add_ray_path(path)
        except TraceError:
            continue
        else:
            entrance_rays.append(path[0])
            exit_rays.append(path[-3])  # -1 and -2 are the world node, don't want that

print("Done!")
```
The error does not occure when size is set to, for example: `size = (1.0, 1.0, 0.02)`

2) The second thing is related to photon counting. First of all, I noticed, that always (also in your examples) photons thrown = 10x defined number of rays. In addition, could you provide me with a small adivce on how to exclude photons passing through unabsorbed from total count of photons exiting LSC?

Thank you for your time. I look forward to your answer.

Kind Regards,
Konrad

**Comment by danieljfarrell** [comment]
Hello Konrad,

Seems like you found some bugs! Thanks.

I will load this script and try and figure out what is going wrong.

What version of pvtrace are you using?

**Comment by Qudzu** [comment]
Hi Daniel,

Thank you for your swift reply.

I use PVTRACE 2.0

All the best,
Konrad

**Comment by danieljfarrell** [comment]
Can you run the command 

    pip show pvtrace

In your terminal and let me know. It guess you are using the latest version. Which should be 2.0.3

**Comment by Qudzu** [comment]
Ah, yes, indeed, it is 2.0.3.

All the best,
Konrad

**Comment by danieljfarrell** [comment]
Konrad,

This is now fixed in pvtrace 2.0.4, updated using,

    pip install pvtrace --upgrade

The problem was that pvtrace was enforcing an accuracy check that was a bit too strict (df1eca638).

However, I cannot reproduce the second comment you made about generating too many rays. If you make a script that shows this then please reopen this issue.

Thanks for the help in making pvtrace better!
